presto-docs/src/main/sphinx/sql/show-stats.rst

```diff
@@ -0,0 +1,31 @@
+==========
+SHOW STATS
+==========
+
+Synopsis
+--------
+
+.. code-block:: none
+
+    SHOW STATS FOR table
+    SHOW STATS FOR ( SELECT * FROM table [ WHERE condition ] )
+
+Description
+-----------
+
+Returns stats for the named table or for the results of a (limited) query.
```

Comment:
> for the results of a (limited) query

this isn't true. This is always stats for a table, as returned by connector, probably in response to pushed down predicate. 

I don't remember what happens if the `condition` cannot be wholly pushed down? Does SHOW STATS fail?

Reply:
If the stats are just for the table, should we still document the SELECT form  if it is really just going to return the same thing as SHOW STATS for table?

Reply:
@findepi @arbass22 SHOW STATS works correctly for filters that use partition keys, e.g. the stats displayed represent the fraction of the data matching the filter. SHOW STATS doesn't fail if the filter includes non-partition keys, but it shows the same results as if these filters were not present. This is confusing. Piotr, do you happen to know if this is by design?

Reply:
It is because predicates are passed to connector to retrieve the data layout. Statistics that are created are based on this layout.

I think we should either make SHOW STATS to:
 - fail in case that layout has some not enforced predicate
 - or make SHOW STATS to be generic and accept any kind of query (with joins etc)